File: yuva_raithu_backend/src/main/java/com/yuvaraithu/backend/models/Product.java
package com.yuvaraithu.backend.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.List;

@Entity
@Table(name = "products")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private Double price;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Category category;

    @Column(length = 2000)
    private String description;

    @Column(length = 1000)
    private String usageInstructions;

    @Column(length = 1000)
    private String benefits;

    private Integer availableStock;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "dealer_id", nullable = false)
    private User dealer;

    @ElementCollection
    @CollectionTable(name = "product_images", joinColumns = @JoinColumn(name = "product_id"))
    @Column(name = "image_url")
    private List<String> imageUrls;

    private Double averageRating = 0.0;
    private Integer totalReviews = 0;
}

File: yuva_raithu_app/lib/features/marketplace/domain/product.dart
class Product {
  final int id;
  final String name;
  final String brand;
  final double price;
  final String category;
  final String description;
  final String usageInstructions;
  final String benefits;
  final int availableStock;
  final String dealerName;
  final int dealerId;
  final List<String> imageUrls;
  final double averageRating;
  final int totalReviews;

  Product({
    required this.id,
    required this.name,
    required this.brand,
    required this.price,
    required this.category,
    required this.description,
    required this.usageInstructions,
    required this.benefits,
    required this.availableStock,
    required this.dealerName,
    required this.dealerId,
    required this.imageUrls,
    required this.averageRating,
    required this.totalReviews,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      name: json['name'] ?? '',
      brand: json['brand'] ?? '',
      price: json['price']?.toDouble() ?? 0.0,
      category: json['category'] ?? '',
      description: json['description'] ?? '',
      usageInstructions: json['usageInstructions'] ?? '',
      benefits: json['benefits'] ?? '',
      availableStock: json['availableStock'] ?? 0,
      dealerName: json['dealerName'] ?? '',
      dealerId: json['dealerId'] ?? 0,
      imageUrls: List<String>.from(json['imageUrls'] ?? []),
      averageRating: json['averageRating']?.toDouble() ?? 0.0,
      totalReviews: json['totalReviews'] ?? 0,
    );
  }
}
